src/main.rs

```diff
@@ -54,6 +57,16 @@ pub fn main() {
         return;
     }
 
+    if let Some(pos) = env::args().position(|a| a == "--explain") {
+        if let Some(mut lint) = env::args().nth(pos + 1) {
+            lint.make_ascii_lowercase();
+            explain(&lint.strip_prefix("clippy::").unwrap_or(&lint).replace('-', "_"));
+        } else {
+            show_help();
```

Comment:
I feel like this should actually say that a lint name was expected, but that is very nit picky and the help message works well enough. :upside_down_face:

Reply:
Like "error: expected lint name. Please use with `cargo clippy -- --explain <lint>`"?

Reply:
Yeah, maybe, but I don't think this is a blocker. Having the help message is good  enough IMO :)